ukis_pysat/data.py

```diff
@@ -38,24 +40,40 @@ class Source:
     Remote APIs and local data directories that hold metadata files are supported.
     """
 
-    def __init__(self, datahub, datadir=None, datadir_substr=None):
+    def __init__(self, datahub, catalog=None, url=None):
         """
         :param datahub: Data source (<enum 'Datahub'>).
-        :param datadir: Path to directory that holds the metadata if datahub is 'File' (String).
-        :param datadir_substr: Optional substring patterns to identify metadata in datadir if datahub
-            is 'File' (List of String).
+        :param catalog: Only applicable if datahub is 'Directory'. Can be one of the following types:
+                        Path to STAC Catalog file catalog.json (String, Path).
+                        Pystac Catalog or Collection object (pystac.catalog.Catalog, pystac.collection.Collection).
+                        None initializes an empty catalog.
+                        (default: None)
+        :param url: Only applicable if datahub is 'StacApi'. STAC Server endpoint, reads from STAC_API_URL environment
+                        variable by default
+                        (default: None)
         """
         self.src = datahub
 
-        if self.src == Datahub.File:
-            if not datadir:
-                raise AttributeError(f"'datadir' has to be set if datahub is 'File'.")
+        if self.src == Datahub.Directory:

```

Comment:
Not sure anymore if "Directory" is a good name for this source. It is more like a static catalog, so perhaps "Catalog"? Loading items from a directory is currently a method that can be applied to any existing catalog.